cli/src/main/java/com/devonfw/tools/ide/common/JsonVersionItem.java

```diff
@@ -0,0 +1,8 @@
+package com.devonfw.tools.ide.common;
+
+/**
+ * Interface for a data object that can be read from and written to JSON.
+ */
+public interface JsonVersionItem {
+
+}
```

Comment:
I have used this to allow for a common implementation of `collectVersionsWithDownloadsFromJson` in `JsonUrlUpdater`, so that each subclass does not have to implement it themselves. 

The `JsonUrlUpdater` is now based on the subclasses `JsonObject` and `JsonVersionItem`, with the first representing the whole json to parse the versions from, and the latter the individual items that contain the versioning number and possibly download links. Each subclass implements their own method of the relation between `JsonVersionItem`s (`JVI`) and the `JsonObject` (`J`) by overriding 
`protected abstract Collection<JVI> getVersionItems(J jsonObject)`